Specs/Core/TimeIntervalCollectionSpec.js

```diff
@@ -643,6 +643,288 @@ defineSuite([
+                { sec: 2, data: 2 },
+                { sec: 4, data: 4 },
+                { sec: 6, data: 6 }
+            ]
+        );
+        expectCollection(
+            collection,
+            4,
+            [
+                { sec: 0, data: 0 },
+                { sec: 1, data: 0 },
+                { sec: 2, data: 2 },
+                { sec: 3, data: 2 },
+                { sec: 4, data: 4 },
+                { sec: 5, data: 4 },
+                { sec: 6, data: 6 },
+                { sec: 7, data: 6 },
+                { sec: 8, data: 6 },
+                { sec: 9, data: 6 },
+                { sec: 10, data: 6 },
+                { sec: 11, data: null }
+            ]
+        );
+
+        addIntervals(
+            collection,
+            [
+                { sec: 1, data: 1 },
+                { sec: 3, data: 3 }
+            ]
+        );
+        expectCollection(
+            collection,
+            4,
+            [
+                { sec: 0, data: 0 },
+                { sec: 1, data: 1 },
+                { sec: 2, data: 1 },
+                { sec: 3, data: 3 },
+                { sec: 4, data: 3 },
+                { sec: 5, data: 3 },
+                { sec: 6, data: 3 },
+                { sec: 7, data: 3 },
+                { sec: 8, data: 6 },
+                { sec: 9, data: 6 },
+                { sec: 10, data: 6 },
+                { sec: 11, data: null }
+            ]
+        );
+
+        addIntervals(
+            collection,
+            [
+                { sec: 3, data: 31 }
+            ]
+        );
+        expectCollection(
+            collection,
+            4,
+            [
+                { sec: 0, data: 0 },
+                { sec: 1, data: 1 },
+                { sec: 2, data: 1 },
+                { sec: 3, data: 31 },
+                { sec: 4, data: 31 },
+                { sec: 5, data: 31 },
+                { sec: 6, data: 31 },
+                { sec: 7, data: 31 },
+                { sec: 8, data: 6 },
+                { sec: 9, data: 6 },
+                { sec: 10, data: 6 },
+                { sec: 11, data: null }
+            ]
+        );
+
+        removeInterval(collection, 3, 8);
+        expectCollection(
+            collection,
+            3,
+            [
+                { sec: 0, data: 0 },
+                { sec: 1, data: 1 },
+                { sec: 2, data: 1 },
+                { sec: 3, data: null },
+                { sec: 4, data: null },
+                { sec: 5, data: null },
+                { sec: 6, data: null },
+                { sec: 7, data: null },
+                { sec: 8, data: null },
+                { sec: 9, data: 6 },
+                { sec: 10, data: 6 },
+                { sec: 11, data: null }
+            ]
+        );
+
+        removeInterval(collection, 0, 1);
+        expectCollection(
+            collection,
+            2,
+            [
+                { sec: 0, data: null },
+                { sec: 1, data: 1 },
```

Comment:
Also note that this assertion is wrong.  After removing `[0, 1]` then the interval collection should not contain `1`.  Just mentioning because you said this was based on other test code you had which presumably has the same error.

Reply:
Nice catch! I finally understand what isStartIncluded means in the TimeInterval! Thanks